Impune webserver

impune-pl opened the pull request:
Simple but somewhat configurable webserver.

impune-pl committed:
Started work on webserver.
Configuration model and loader are complete and working.
Created sample configuration file explainig what parameters are for.

impune-pl committed:
Turns out there is no need to switch port

impune-pl committed:
Work on webserver in progress.
Decided to use python's built-in threading http server instead of
implementing own version.
Added handler for http basic auth, untested.
 Added configuration keys for logging and some basic logging.

impune-pl committed:
Changed config loader to load configuration into
Configuration instance instead of making some weird pseudo class on the fly

mateusz-szczyrzyca commented on .vscode/launch.json:
```diff
@@ -0,0 +1,16 @@
+{
```
To chyba nie jest plik projektu? :)

impune-pl replied:
Konfiguracja do uruchamiania debuggera. Można się kłócić że przydatna, ale fakt, że nie powinna się tu znaleźć. :-D

impune-pl replied:
Usunąłem

impune-pl committed:
Update .gitignore and remove vs code configuration

impune-pl committed:
Basic functionality of displaying index file in
 response to request to '/' works. Other functionalities
 I'll test soon.

mateusz-szczyrzyca commented:
Wygląda legitnie, spróbuję w tym tygodniu jeszcze stestować :)

impune-pl commented:
Stestowałem wczoraj, zamiast wyświetlać listę plików dostaję index pod każdą ścieżką - coś jest nie tak z rozpoznawaniem. Dzisiaj się przyjrzę co się dzieje.

impune-pl committed:
All basic functionalities seem to work fine.
Still need to work on regex matching for index files.
Did not check if "list_files": false works yet.
Access restriction is not tested yet.

impune-pl committed:
Documentation and categorized list of tasks added.
Gitignore updated to ignore images used for tests.

impune-pl committed:
Update exceptions to get rid of path parameter.

impune-pl committed:
Remove empty line from top of file.
Replace calls to os.path.join with calls to fsh.concat_paths.
Optimize imports.

impune-pl committed:
Add else for readability.

impune-pl commented on server/files/www_files_helper.py:
```diff
@@ -0,0 +1,46 @@
+    contents, len = fsh.get_file_contents(os_path)
+    mimetype = fsh.get_file_mimetype(os_path)
+    if mimetype is None:
+        mimetype = "text/html;charset=utf-8"
```
@impune-pl prawdopodobny winowajca buga z tekstowymi obrazkami.
Może jakaś biblioteka określająca mimetype na podstawie zawartości pliku byłaby dobrym fallbackiem.

impune-pl commented on server/request_handlers.py:
```diff
@@ -20,75 +18,23 @@ def __init__(self, request: bytes, client_address: tuple[str, int], server: Thre
-        f.seek(0)
+            file = wfh.get_file_descriptor_for_path(os_path, self._config)
+        except PathOutOfScopeError:
+            return self.respond_with_not_found()
```
@impune-pl forbidden może być lepszą opcją w tym wypadku.

mateusz-szczyrzyca commented:
Możemy omówić ten program przy następnym callu projektowym, zobaczymy czy działa, jakie ma problemy i ew. metodą pair/mob programming, spróbujemy poprawić aby inni mogli współuczestniczyć

impune-pl committed:
Basic auth and mimetype recognition work correctly
Response when path points at something outside configured www catalog
changed to Forbidden

impune-pl committed:
Added and tested regex-based index files selection
